### ISSUE TITLE
Lint rule for spaces surrounding apostrophes (')

### ISSUE THREAD

**Issue by ericxiao251** [sqlfluff/sqlfluff#360]
Can we have a lint rule that checks for spaces surrounding apostrophes? Following Simon Holywell's SQL Style Guide:

> Although not exhaustive always include spaces:
> * ...
> * surrounding apostrophes (') where not within parentheses or with a trailing comma or semicolon.

reference: https://www.sqlstyle.guide/#white-space

**Comment by alanmcruickshank** [comment]
Agreed.

Not for implementation of this one: rather than looking for the `'` or `"` characters literally, we should look for `StringLiteral` segments which then make sure that things are common between dialects that use different quote characters.

This would also be a great starter rule if anybody wanted to have a go at it.

**Comment by ericxiao251** [comment]
I can take a stab at it :).

**Comment by pwildenhain** [comment]
I like this rule too (once I wrapped my head around what it was checking for 💫 )

For anyone else reading this, and having a brainfart, this rule would help turn this:
🙅 
```sql
select
    col1,
    'something'as col2
from my_table
```
into this 🙌 
```sql
select
    col1,
    'something' as col2
from my_table
```

`sqlfluff` currently *does not* complain about the former.

@ericxiao251 I might be interested in tackling this too, though I will defer to you if you already started working on something

**Comment by ericxiao251** [comment]
go for it, I haven't started yet!

**Comment by pwildenhain** [comment]
Great! Here is another good test case:

```sql
SELECT
    a AS"col"
FROM c
```

**Comment by NiallRees** [comment]
I think this relates to #607 if we can crack two in one!

**Comment by alanmcruickshank** [comment]
I've started a solution for this one, gives me an opportunity to refactor L006 at the same time, because it *should* be a drop in replacement for this (and unfortunately it's not 😒 ).
